h_malloc.c

```diff
@@ -99,6 +104,18 @@ struct slab_metadata {
 #if SLAB_QUARANTINE
     u64 quarantine_bitmap[4];
 #endif
+#ifdef HAS_ARM_MTE
+    // arm_mte_tags is used as a u4 array (MTE tags are 4-bit wide)
+    //
+    // Its size is calculated by the following formula:
+    // (MAX_SLAB_SLOT_COUNT + 2) / 2
+    // MAX_SLAB_SLOT_COUNT is currently 256, 2 extra slots are needed for branchless handling of
+    // edge slots in tag_and_clear_slab_slot()
+    //
+    // It's intentionally placed at the end of struct to improve locality: for most size classes,
+    // slot count is far lower than MAX_SLAB_SLOT_COUNT.
+    u8 arm_mte_tags[129];
+#endif
```

Comment:
```suggestion
#ifdef HAS_ARM_MTE
    #define MAX_SLAB_SLOT_COUNT 256
    // arm_mte_tags is used as a u4 array (MTE tags are 4-bit wide),
    // and is intentionally placed at the end of struct to improve locality: for most size classes,
    // slot count is far lower than MAX_SLAB_SLOT_COUNT.
    //
    // The 2 extra slots are needed for branchless handling of
    // edge slots in tag_and_clear_slab_slot()
    u8 arm_mte_tags[(MAX_SLAB_SLOT_COUNT + 2) / 2];
#endif
```

Reply:
There are several hard-coded constants that define and depend on `MAX_SLAB_SLOT_COUNT` (`bitmap` and `quarantine_bitmap` in this struct, constants in size class definitions). I've used a define initially, but changed back to a hardcoded constant for consistency.

Reply:
I think a define is better, if only to keep things in sync.

Reply:
I think using a literal makes size of this struct (and overhead of adding arm_mte_tags array) more obvious. 
I think that defining a constant would imply that its value can be easily adjusted, which is actually hard to do properly.